edx_sysadmin/api/test_decorators.py

```diff
@@ -0,0 +1,82 @@
+"""
+Tests for views
+"""
+import pytest
+import json
+
+from hashlib import sha256
+import hmac
+
+from django.conf import settings
+from django.http import HttpResponse
+from django.test import TestCase, RequestFactory, override_settings
+from django.urls import reverse
+from django.utils.encoding import force_bytes
+from rest_framework import status
+
+from edx_sysadmin.api.decorators import authenticate_github_request
+
+
+class AuthenticateGithubRequestTestCase(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.factory = RequestFactory()
+
+    def test_authenticate_github_request_without_key(self):
+        """Test authenticate_github_request decorator without signature key"""
+
+        request = self.factory.post(reverse("sysadmin:api:git-reload"))
```

Comment:
In these tests I think we can simply use restframework's [apiclient](https://www.django-rest-framework.org/api-guide/testing/#apiclient) to hit our API instead of creating requests after we have created a customer permission for authenticating github hooks.

Reply:
Updated.